scripts/saveState.js

```diff
@@ -65,8 +65,9 @@ function addToSandwich(type, item){
             
         }
 		sidebarBody().find("#sideSubgoal" + item.id).unbind( "click" ).click(function(){
+			subArr = getSubgoalArrayFromLocal(); // in case something changes before the button is clicked
 			// do not enter drawSubgoal with a different id until the current subgoal is saved
-			if (statusIsTrue("gotSubgoalQuestions")){
+			if (statusIsTrue("gotSubgoalQuestions") || item.id == subArr.length){
```

Comment:
Turns out this is necessary if you click away to an action (because actions can actually be viewed before saving, unlike subgoals) 😅  In this case, there needs to be a way back to the current subgoal, even if it hasn't been saved. Trouble before was that subArr wasn't up to date, so item.id == subArr.length _always_, making the statement useless.

Reply:
Ahhhhhhh, I see, did you test it and click on actions to see if the right subgoal number was being generated?

Reply:
This a fix for another little bug I noticed, haha. Fix for the overwriting coming soon.